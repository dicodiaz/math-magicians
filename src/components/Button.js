import PropTypes from 'prop-types';
import React from 'react';

// eslint-disable-next-line react/prefer-stateless-function
export default class Button extends React.Component {
  render() {
    const { classes, text } = this.props;
    classes.push('py-4', 'border', 'border-custom1', 'fs-5');
    const classesString = classes.join(' ');
    return (
      <button type="button" className={classesString}>
        {text}
      </button>
    );
  }
}

Button.propTypes = {
  classes: PropTypes.arrayOf(PropTypes.string).isRequired,
  text: PropTypes.string.isRequired,
};
