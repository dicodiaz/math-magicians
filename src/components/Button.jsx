import PropTypes from 'prop-types';

const Button = ({ text, type, onClick }) => {
  const classNameArray = ['py-4', 'border', 'border-custom1', 'fs-5'];
  switch (type) {
    case 'normal':
      classNameArray.push('bg-custom2');
      break;
    case 'operator':
      classNameArray.push('bg-custom3');
      break;
    case 'wide':
      classNameArray.push('bg-custom2', 'col-6');
      break;
    default:
  }
  const className = classNameArray.join(' ');

  return (
    <button type="button" className={className} onClick={onClick}>
      {text}
    </button>
  );
};

Button.propTypes = {
  text: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default Button;
