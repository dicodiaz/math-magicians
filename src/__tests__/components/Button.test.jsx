import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import Button from '../../components/Button';
import { click } from '../test-utils';

describe('Button component', () => {
  const onClick = jest.fn();

  it('should render as expected', () => {
    const { container } = render(<Button text="1" type="normal" onClick={onClick} />);
    expect(container).toMatchSnapshot();
  });

  it('should customize inner text from prop: text', () => {
    const tree = render(<Button text="2" type="normal" onClick={onClick} />);
    const button = tree.getByRole('button', { name: '2' });
    expect(button).toBeVisible();
  });

  it('should customize classes from prop: type = operator', () => {
    const tree = render(<Button text="1" type="operator" onClick={onClick} />);
    const button = tree.getByRole('button', { name: '1' });
    expect(button).not.toHaveClass('bg-custom2');
    expect(button).toHaveClass('bg-custom3');
  });

  it('should customize classes from prop: type = wide', () => {
    const tree = render(<Button text="1" type="wide" onClick={onClick} />);
    const button = tree.getByRole('button', { name: '1' });
    expect(button).toHaveClass('bg-custom2');
    expect(button).toHaveClass('col-6');
  });

  it('should run onClick when clicked', async () => {
    const tree = render(<Button text="1" type="wide" onClick={onClick} />);
    const button = tree.getByRole('button', { name: '1' });
    await click(button);
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
