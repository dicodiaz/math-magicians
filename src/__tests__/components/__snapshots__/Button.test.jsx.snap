// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render as expected 1`] = `
<div>
  <button
    class="py-4 border border-custom1 fs-5 bg-custom2"
    type="button"
  >
    1
  </button>
</div>
`;
